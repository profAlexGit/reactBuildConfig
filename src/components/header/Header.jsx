import React from 'react';
import styles from './header.less';

export const HeaderComponent = () => {
  return (
    <header>
      <h1 className={styles.example}>Header</h1>
    </header>
  );
};

export default HeaderComponent;
