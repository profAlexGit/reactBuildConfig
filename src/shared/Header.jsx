import { hot } from 'react-hot-loader/root';
import React from 'react';
import styles from './header.less';

const HeaderComponent = () => {
  return (
    <header>
      <h1 className={styles.example}>Reddit for our own!!!</h1>
    </header>
  );
};

export const Header = hot(HeaderComponent);
