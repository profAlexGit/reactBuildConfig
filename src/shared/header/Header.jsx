import React from 'react';
import styles from './header.less';

export const HeaderComponent = () => {
  return (
    <header>
      <h1 className={styles.example}>Reddit for our own!!!</h1>
    </header>
  );
};

export default HeaderComponent;
