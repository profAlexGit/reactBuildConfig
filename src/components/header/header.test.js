import React from 'react';
import { configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

import { HeaderComponent } from './Header.jsx';
configure({ adapter: new Adapter() });

describe('<HeaderComponent />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(
      <HeaderComponent />
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<HeaderComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('h1', () => {
    expect(wrapper.find('h1')).toHaveLength(1);
  });
});
