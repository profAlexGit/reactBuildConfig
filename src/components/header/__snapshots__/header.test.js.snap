// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderComponent /> renders correctly 1`] = `
<header>
  <h1>
    Header
  </h1>
</header>
`;
